## Tipos de tasas de interés
En la teoría financiera, las tasas de interés vencidas son fundamentales porque representan el rendimiento de oportunidad de un periodo (futuro). La tasa *nominal* es la tasa referencial, que será útil para calcular la tasa *efectiva*. Por ejemplo, si nos pagan mensualmente una tasa del 12% anual, no significa que la tasa de referencia sea la que vamos a recibir. Lo que sí recibiremos en cambio será una proporción de la tasa anual considerando, en este caso, un mes.   $$i_{m}=\frac{j_{m}}{12}=\frac{.12}{12}=.01$$
$j_{m}$ representa la tasa nominal, mientras que $i_{m}$ es la tasa propocional aplicable al periodo, que en este caso es mensual. La tasa proporcional coincide con la tasa efectiva mensual, es decir, se gana *efectivamente* 1% en un mes. 

In [7]:
def tproporcion (jm,n):
    return jm/n


Sea una tasa anual de 16.6%, la proporcional capitalizable trimestralmente sería 4.15%:

In [6]:
tproporcion(.166,4)

0.0415

La tasa efectiva se puede entender como la tasa que produce el mismo resultado capitalizando una vez de otra tasa a diferentes capitalizaciones. $$(1+i)=(1+\frac{j_{m}}{m})^m$$
$$i=(1+\frac{j_{m}}{m})^m-1$$

En donde $i$ es la tasa efectiva; $j_{m}$, la nominal; $\frac{j_{m}}{m}$, la proporcional y m el número de capitalizaciones.

In [18]:
def tasaefectiva (jm,m):
    return (1+jm/m)**m -1

Por ejemplo, tenemos una tasa anual de 12% capitalizable semestralmente, queremos la tasa en la que es indiferente que capitalice. 

In [20]:
tasaefectiva(.12,2)

0.12360000000000015

Es decir, con una tasa de 12.36% *equivale* a una tasa de 12% anual capitalizable semestralmente.

Para la misma tasa nominal capitalizable mensualmente:

In [26]:
tasaefectiva(.12,12)

0.12682503013196977

La tasa de 12.68% equivale a una tasa del 12% anual capitalizable mensualmente.

Se ofrece una tasa anual de 6% para una operación mensual y queremos calcularla a 3 meses:

In [64]:
(1+tasaefectiva(.06,12)/12)**3-1

0.015498841925007945

La tasa aproximada es de 1.549%.

La tasa equivalente es la que al capitalizar a difentes periodos es igual que la tasa efectiva. Se tiene que $$(1+i_{m})^m=(1+i)$$ y despejando tendríamos $$i_{m}=(1+i)^\frac{1}{m} -1$$

En donde $i$ es la tasa equivalente, $i_{m}$, la tasa efectiva y $m$ el plazo capitalizable. Notar que para calcular la tasa anual efectiva con días es necesario cambiar la base (tasa proporcional/365 y plazo sobre días). 

In [58]:
def tasaequivalente (i,m):
    return (1+i)**(1/m)-1

In [72]:
tasaequivalente(0.1236,12)

0.009758794179192343

Quiere decir que invertir a 0.97% al mes es igual que ganar 12.36% anualmente porque producen el mismo rendimiento.

Es decir, con la tasa nominal calculamos la tasa proporcional, que sirve para obtener la tasa efectiva, que se utiliza para la tasa equivalente. Aunque este orden no importa porque la relación existe siempre, solo es para una interpretación práctica.

## Valor presente y valor futuro

Hay una distinción entre calcular valor presente de flujos vencidos y adelantados. 

Para el primer caso: $$S_{v}=\frac{C}{(1+i)}+\frac{C}{(1+i)^2}+\frac{C}{(1+i)^3}+\dots+ \frac{C}{(1+i)^n}$$

En donde  $S_{v}=\frac{1}{1+i}\frac{1-q^n}{1-q}$, por progresión geométrica y factorizando el factor de descuento.
$q$ es la razón de la progresión, que es $\frac{1}{1+i}$. Sustituyendo tenemos: $$S_{v}=\frac{1}{1+i}\frac{1-\frac{1}{(1+i)^n}}{1-\frac{1}{(1+i)}}$$ Simplificando se tiene el valor presente para flujos vencidos $$V_{v}=C\frac{(1+i)^n-1}{(1+i)^n i}$$

Cuando son pagos adelantados se tiene que el primer pago no se descuenta y finaliza un periodo antes que en el pago vencido: $$S_{a}= C+\frac{C}{(1+i)}+\frac{C}{(1+i)^2}+\frac{C}{(1+i)^3}+\dots+ \frac{C}{(1+i)^{n-1}}$$ Se procede similarmente como en el caso de valor presente vencido: $$S_{a}= C\frac{1-\frac{1}{(1+i)^n}}{1-\frac{1}{(1+i)}}$$ Finalmente se tiene: $$V_{a}=C(1+i)\frac{(1+i)^n-1}{(1+i)^ni}$$

Otras formas equivalentes son $V_{a}=C[\frac{1-(1+i)^{-n}}{i}](1+i)$ y $V_{v}=C[\frac{1-(1+i)^{-n}}{i}]$. En general el valor presente para pagos vencidos es el más común en el cálculo financiero.

In [126]:
def VPa(C,i,n):
    return C*((1-(1+i)**(-n))/i)*(1+i)

In [127]:
def VPv(C,i,n):
    return C*(1-(1+i)**(-n))/i

Sea un flujo de 20 pagos  de $100 semestrales con interes del 3%, calcular el valor presente para pagos vencidos y pagos anticipados.

In [128]:
VPv(100,.03,20), VPa(100,.03,20)

(1487.7474860455518, 1532.3799106269182)

Intuitivamente el valor presente para pagos anticipados es mayor 
porque el primer pago no se descuenta a la tasa de referencia.

El valor futuro nos dice cuánto dinero tendremos en un lapso futuro dado un capital y una tasa de interés. $$S_{v}= C+C(1+i)+C(1+i)^2+C(1+i)^3+\dots + C(1+i)^n$$
De forma similar al valor presente: $$S_{v}=C\frac{q^n-1}{q-1}= C\frac{(1+i)^n-1}{(1+i)-1}$$
Y el valor futuro para pagos vencidos es $$VF_{v}=C\frac{(1+i)^n-1}{i}$$ 

En el caso de pagos anticipados es como sigue:

$$VF_{a}=VF_{v}(1+i)$$
 que significa, como en valor presente, que se vuelve a capitalizar de forma simple el monto vencido porque se da por atrasado el ahorro, lo cual genera más capitalizaciones.

En un ahorro interesa saber cuánto se va a juntar de aquí a 10 años convertible semestralmente dando $100 y a una tasa semestral efectiva de 3%. Es de interés saber cuál conviene más: ahorrar vencidamente o anticipadamente.

In [3]:
def VFv(C,i,n):
    return C*(((1+i)**n-1)/i)

In [4]:
def VFa(C,i,n):
    return C*(1+i)*(((1+i)**n-1)/i)

In [5]:
VFv(100,.03,20), VFa(100,.03,20)


(2687.0374488980497, 2767.648572364991)

Por lo que el ahorro con pagos anticipados genera mayores cifras y conviene más.

## Rentas perpetuas

Es una serie de pagos que no vence. Se dice que su número de periodos tiende a infinito. No se puede calcular el valor futuro porque sería una serie divergente pero sí es posible calcular el valor presente. En una renta de este tipo solo se conoce la tasa de interés, su momento de inicio y los pagos. 

Existen rentas perpetuas inmediatas, diferidas y anticipadas. 

Veamos el primer caso, renta perpetua inmediata de pagos vencidos, tenemos que: $$S=\frac{C}{(1+i)}+\frac{C}{(1+i)^2}+\frac{C}{(1+i)^3}+\dots \infty$$

Viendo que es una progresión geométrica (como en valor presente y valor futuro) se llega a que $$S=\frac{C}{(1+i)}\frac{1-\frac{1}{(1+i)^n}}{1-\frac{1}{(1+i)}}$$

para $$\lim _{n \to \infty} \frac{C}{(1+i)}\frac{1-\frac{1}{(1+i)^n}}{1-\frac{1}{(1+i)}}= \frac{C}{(1+i)}\frac{1}{\frac{(1+i)-1}{(1+i)}}= \frac{C}{i} $$

In [9]:
def perpevencida(C,i):
    return C/i

Sea una renta perpetua de 1 con tasa de 10%. El valor de la renta perpetua es 10. Tiene sentido porque a plazo largos el valor de la renta vale cero.

In [8]:
perpevencida(1,.1)

10.0